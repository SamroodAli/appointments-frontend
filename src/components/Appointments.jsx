import getAppointments from '../api/getAppointments';
import useLockedRoute from '../hooks/useLockedRoute';

const Appointments = () => {
  const {
    isLoading, isError, data, error,
  } = useLockedRoute('appointments', getAppointments);

  if (isLoading) {
    return <span>Loading...</span>;
  }

  if (isError) {
    return (
      <span>
        Error:
        {error.message}
      </span>
    );
  }

  return (
    <ul>
      {data.data.map((appointment) => (
        <li key={appointment.id}>
          {appointment.teacher_id}
        </li>
      ))}
    </ul>
  );
};
export default Appointments;
