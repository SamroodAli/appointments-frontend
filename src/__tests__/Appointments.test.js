import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Appointments from '../components/Appointments';
import Wrapper from '../testWrapper';

test('Appointments snapshot testing', () => {
  const tree = renderer.create(
    <Wrapper>
      <Appointments />
    </Wrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
