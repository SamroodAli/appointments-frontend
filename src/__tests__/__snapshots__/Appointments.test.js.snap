// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments snapshot testing 1`] = `
<button
  className="bg-green-500 p-2  top-0 w-full"
  disabled={true}
  type="button"
>
  <p
    className="font-mono animate-pulse"
  >
    Trying to log you in
  </p>
</button>
`;
